### DISCUSSION TITLE
Warum braucht es eine Bundes-Wallet?

### DISCUSSION THREAD

**Discussion by ii-nik** [e-id-admin/governance-sounding-board#12] [Q&A] [answered]
Ich bin mir selber nicht sicher, darum frage ich das hier einmal.

Unter der Annahme, dass die eID nur ein VC ist, könnten wir uns auch auf den Standpunkt stellen, dass ein Owner / Holder diese in seine Wallet App seiner Wahl ablegen können soll.

Vorteil einer Bundes-Wallet
* Die Bundes-Wallet wäre sicher in Bezug auf die Speicherung der eID, weil konsequent die TPM Funkitonalitäten benutzt werden

Nachteil einer Bundes-Wallet
* Nicht alle Menschen werden einer Bundeswallet vertrauen, obwohl der Sourcecode (hoffentlich) offengelegt würde (Erfahrung aus Covid-Zert App). 

Vorteile freie Wahl des Wallet
* Ist UX mässig näher an der Realität: In meinem Portmonnaie habe ich auch ganz unterschiedliche Arten von Nachweisen und auch Geld (das eigentlich auch nur ein Nachweis ist  ;-))
* Der Mensch kann seine Wallet von einem Anbieter wählen, zu dem er das grösste Vertrauen hat. Die Akzeptanz ist in diesem Fall höher, resp. man erreicht mehr Menschen

Nachteile freie Wahl des Wallet
* Die Sicherheit kann nicht staatlich sichergestellt werden. Es könnte unsichere Wallet Apps geben.
* Missbrauchspotential ist höher. (Aber die gibts auch mit der Plastik-ID, auch dort geht die Verantwortung der sicheren Aufbewahrung mit der Übergabe der Plastik-ID an den Menschen über)

Für mich stellt es sich so dar, dass wir nicht "s'Füfi und s'Weggli" haben können.

Daher meine vorläufige Haltung: Bauen wir mal ein Bundes-Wallet als Prototyp, stellen aber gleichzeiti sicher, dass nur Standards verwendet werden, welche auch von anderen (z.T. bereits erhältlichen) Wallets unterstützt werden können (z.B. DIDComm).

Was denkt ihr? Was denken Sie?

**Comment by KampfCaspar** [comment]
Sämtliche Definitionen (Kommunikation/Formate) muss offen sein und es darf keine Einschränkung des genutzten Wallets geben - meine Daten meine Verfügungsgewalt. 

Es gibt eine Referenz-Wallet des Bundes (Opensource). Diese ist im Idealfall so gut, dass sie auch von der überwiegenden Mehrheit verwendet wird.

**Comment by grt-dti** [comment, marked as answer]
Da die Sicherheit des Wallet DAS massgebliche Kriterium für den rechtssicheren Einsatz der E-ID ist, müsste man sich eigentlich folgendes überlegen: Würde ich meine Plastik-ID in ein fremdes Portemonnaie legen, wenn ich mir nicht sicher darüber sein kann, ob der Hersteller des selbigen nicht einfach eine gültige(!) Kopie davon ziehen und somit in meinem Namen handeln kann?

Elektronische Informationen können beliebig kopiert werden, solange man nicht einen grossen physikalischen Aufwand betreibt genau dieses zu verhindern. In dem Augenblick, da ich Backup- und Recoveryfunktionen erlaube, erlaube ich Kopien der elektronischen Information und zwar dummerweise sogar absolut nicht-unterscheidbare Kopien - der vorherige ("grosse physikalische") Aufwand ist nutzlos.

WENN der Staat also für die Richtigkeit / den rechtsverbindlichen Einsatz der E-ID garantieren soll (also rechtliche Verbindlichkeit sicherstellen soll) und dabei Backup- und Recoveryfunktionen der E-ID möglich sein müssen, MUSS er die Kontrolle über das Wallet behalten. Man kann Backup & Recovery von der Wunschliste streichen und erhält Freiheit beim Wallet; man kann die Freiheit des Wallet streichen und erhält Backup&Recovery; man kann auch den rechtssicheren Einsatz (die Garantie des Staates) streichen und bekommt beide Freiheiten - aber dann bin ich keinen Schritt weiter als heute......

Wir müssen uns eine wirklich wichtige Frage stellen: Wem vertrauen wir bei unserer staatlichen Identität am meisten? Dem sollten wir die ID und die Speicherung selbiger anvertrauen......

**Reply by ChristophGraf-SWITCH** [reply]
Die "Kontrolle über das Wallet" muss nicht zwingend bedeuten, dass nur ein "staatliches Wallet" infrage kommt. Auf Basis klarer Requirements könnten auch Wallets Dritter zertifiziert und freigegeben werden.
Man könnte sich auch überlegen, ob abgestufte Sicherheit möglich ist. Wichtig hier ist, dass dem Verifier die Information vorliegt, welche Sicherheitsstufe vorliegt. Also ob der Proof z.B. von einem zertifizierten Wallet ausgestellt wurde oder nicht.

**Reply by grt-dti** [reply]
Und wie das RICHTIG ist! Leider löst eine Zertifizierung das Problem der freien Wallet-Wahl nur marginal......

**Reply by fforootd** [reply]
Die Sicherheit des Wallets ist ein grosser Punkt da gebe ich dir recht. 

Allerdings liese sich auch argumentieren das ein "Staatliches Wallet" nur marginal von der Idee eines zentralisierten IdP entfernt ist. Schlussendlich kann der Anbieter des Wallets die Telemetriedaten auch dezentral erhebn und zentralisiert raportieren. 

OSS schütz davor wenig. Da in den grossen Appstores für Kunden nicht nachvollziehbar ist ob auch tatsächlich der "richtige" code auf ihrem device läuft.

**Reply by grt-dti** [reply]
korrekt; das Wallet ist der neuralgische Punkt. Ist das Ding nicht in allen Sicherheitszielen vertrauenswürdig, bringt die ganze schöne SSI-Idee nix. Und diese Wallet-Software wollen wir auf Treibsand bauen und in einem undurchdringlichen Geflecht an Softwarekomponenten, Bibliothekchen, Compilern, Linkern und IDE's entwickeln - wie soll das den vielen Sicherheitszielen gerecht werden können?
Am Ende muss es aber eine "dem Ausweis gleichgestellte" Lösung werden - viel Spass. In den Plastik-Ausweis (quasi die Ausweis-App) hat die Schweiz 50 Jahre Entwicklungsarbeit und zig-Millionen Entwicklungskosten investiert.....

**Reply by fforootd** [reply]
Einverstanden. Aktuell ist einfach weder die ganze VC noch SSI Welt Matur für so ein Unterfangen vor 2025.

IMO Auch wenn der Deutsche Perso keine Verbreitung hat ist dieser meiner Meinung nach der beste tradeoff denn es aktuell gibt.

**Reply by grt-dti** [reply]
für das anvisierte Anwendungsspektrum ja, aber den können wir nicht einfach kopieren.
Der nPA ist schon fast eine SSI (auch wenn mich die SSI-Jünger dafür jetzt steinigen möchten). Das Ding ist dezentral, es ist extrem datensparsam und der Anwender/Inhaber hat es maximal unter seiner Kontrolle. Muss ein VC eigentlich immer ein JSON-Token sein oder kann man ein VC auch "anders" strukturieren/repräsentieren?
Würde mich mal interessieren, wie der nPA unter den SSI-Prinzipien abschneidet - hat das schon mal jemand abgeklopft?

**Comment by ii-nik** [comment]
Aber auch mit einer Bundeswallet ist Backup / Recovery nicht ohne, falls das Recovery auch an ein anderes Gerät gehen soll (Mobile Phone kaputt, gestohlen, etc). Dazu muss das Wallet geöffnet übertragen werden, da private Keys nicht vom TPM des einen in das TPM des anderen Mobile Phones kopiert werden können. Siehe https://developer.apple.com/documentation/security/certificate_key_and_trust_services/keys/storing_keys_in_the_secure_enclave für IOS, Android vermutlich analog. 

Daher: In einem solchen Fall müsste durch die Behörde nach der Prüfung der Person und Mobile Phone ein neues VC ausstellen.

Oder aber, wir machen auch bei der Bundeswallet Abstriche bei der Sicherheit, zugunsten besserer UX und User Akzeptanz.

Dass Menschen auf ihre digitalen Assets achtgeben müssen ist lernbar. Viele der jetzigen Bedenken gab es vor ca 30 Jahren bei der Einführung von e-Banking und heute können wir grösstenteils damit gut umgehen.

**Reply by ChristophGraf-SWITCH** [reply]
Wenn wir Pass oder ID-Karte verlieren, so ist das eine mühsame Sache. Ich denke es ist absolut lernbar, dass gewisse unserer digitalen Assets einen ähnlichen Schutz brauchen und im Fall eines Verlustes ähnlich mühsame Konsequenzen provozieren.

Diese Analogie dünkt mich verständlich und gut kommunizierbar.

Zudem sollte wohl noch die Option ausgelotet werden, ob abgestufte Sicherheit allenfalls nützlich sein könnte. So wie heute ab und an ein Scan einer ID ausreicht, so könnte in Analogie dazu allenfalls auch ein Wallet mit tieferem Schutz und etwas einfacheren Prozessen angeboten werden.

**Reply by KampfCaspar** [reply]
Solange ich verschiedene physische Objekte habe, kann ich je nach 'Gefährdung' nur einzelne mitnehmen - so verliere ich im Ausgang vielleicht nur die ID und nicht den Führerausweis.

Mit eDokumenten in _einer_ Wallet würde ich zwingend _alles_ verlieren - und wenn man dann noch beliebige Dokumente im Ökosystem zulässt, noch schlimmer. Die Situation ist deshalb nicht mit dem Verlust der 'normalen' ID vergleichbar! Eine Backup-Möglichkeit bleibt IMHO absolut zwingend.

Wirklich sicherheitsrelevant ist aber nur die Kern-ID - mit abhängigen Dokumenten kann ich mich nicht alleine ausweisen. Man könnte diese Kern-ID deshalb vom Backup ausnehmen. Dann würde ich spezifisch für die Kern-ID aber zusätzlich ein physisches e-Element vorsehen (z.B. NFC in physischer ID), mit dem ich als 'Vater-Element' eine neue eID im Wallt initialisieren kann.

**Reply by grt-dti** [reply]
ganz generell ist Backup&Recovery gar kein so grosses Problem WENN das Recovery vom Original zweifelsfrei zu unterscheiden ist. Das würde bedeuten, dass beim Recovery ein neues VC ausgestellt werden müsste. Wenn man dann beim Recovery das originale VC gleich noch zurückziehen kann, macht das die ganze Geschichte noch verlässlicher.
Entscheidend sind aber zwei Dinge
a. das Schlüsselmaterial wird immer auf dem SE/TPM generiert
b. man kann kein Schlüsselmaterial in die Wallet "von aussen" einbringen
--> genau dafür braucht es eine "Bundeswallet" - was nicht heisst, dass es genau dieses Eine geben muss; eine Zertifizierung würde problemlos ausreichen.
WICHTIG ist aber, dass man die Verbindungen zwischen zwei VC's nicht kryptographisch verschränkt, weil dann das Recovery des Schlüsselmaterials notwendig würde. Allerdings sollte man eh immer davon ausgehen, dass Schlüsselmaterial alle paar Jahre aufgefrischt werden muss und so würde eine solche Verschränkung bei der "Auffrischung der Schlüssel" zu einem Spiessrutenlauf.....

In einem Punkt bin ich nicht deiner Meinung: Die Sicherheitsbedenken beim e-Banking sind nicht verschwunden, weil sich eine Awareness beim Anwender breit gemacht hat, sondern weil die Banken das Restrisiko tragen müssen, wenn mit ihren "digitalen Assets" was schief geht. Bis auf einen kleinen Pauschalbetrag, bleibt die Bank auf den Kosten sitzen......

**Comment by ii-nik** [comment]
Ja, aber ein neues VC kann nur vom Issuer ausgestellt werden. Das ist möglicherweise mühsam ... Bin hier leider nicht mehr ganz auf dem Laufenden, aber wenn ich mich richtig erinnere, wurden solche "Refresh-Prozesse" zumindest in der Indy Ecke zumindest mal diskutiert. 

Die Diskussion bringt mich auf Gedanken.... Was wäre, wenn wir gar keine Bundeswallet auf dem Mobile Phone machen würden, sondern wenn auf dem Mobile Phone nur der Agent sitzen würde und ich meine Wallet eigentlich in der Cloud beim Provider meines Vertrauens hätte? Ich könnte dann von mehreren Mobile Phones und auch vom PC aus diese Cloud Wallet benutzen ....

**Reply by KampfCaspar** [reply]
Grosse Gefahr des Datenabgreifens. Möglichst nicht, IMHO. Dann lieber ein standardisierter Prozess, um ein 'physisches VC' wieder in die Wallet zu bekommen ohne beim Issuer vorbeizugehen (à la NFC in physischer ID -> eID).

**Reply by christianheimann** [reply]
Nach meiner Kenntnis verfolgt das [Findy-Projekt](https://findy.fi/en/) in Finnland einen Ansatz in diese Richtung. Die Sicherheit wird dabei so gedacht, dass der Zugang mit einem oder mehreren FIDO-Token (Hardware) möglich ist. Solange man also eines dieser Token besitzt, kann man beliebige neue hinzufügen. Verliert man das letzte, so sind die Inhalte "verloren".

**Reply by grt-dti** [reply]
hmmm; das klingt sexy, verschiebt den Konzeptfokus aber wieder ein wenig in Richtung der IdP-Idee (ID liegt beim Cloud-Provider und ich habe "nur" ein Credential um darauf Zugriff zu erhalten).
Die Idee von @KampfCaspar tönt fast noch "besser": was wäre, wenn wir die ID auf dem Natel nur als ge'cache'te Kopie der Plastik-ID beschreiben würden? Wieso nicht "alle x Monate" den ICAO neben's Natel legen, die Kopie auffrischen (z.B. in dem man sich beim fedpol ein neues VC für weitere x Monate holt) und für die ganzen 0815-Einsatzszenarien benutzen.
Wenn's dann wirklich mal "an's Eingemachte geht" (QES, EPD,...) dann reicht die Kopie nicht, dann muss es der echte Ausweis sein, der dann direkt abgefragt und ein nur x Minuten gültiges VC ausstellt.
Vorteil des Ganzen: man müsste das Wallet nicht über alle Massen schützen, weil die Kopie eben doch nur eine Kopie ist und als solche erkennbar......

**Reply by fforootd** [reply]
Also von der Idee VCs auf einem Server (einer zusätzlichen Partei) zu speichern bin ich kein Freund. Damit erhöht man die Angriffsoberfläche erheblich. Der Anbieter oder ein Angreifer dieses Servers könnte auch ohne weiteres eigene FIDO Keys hinterlegen. Das Problem einer möglichen Impersonation nimmt damit zu.

**Reply by grt-dti** [reply]
das klappt ausschliesslich als Datenspeicherort - beim Cloudanbieter darf es keine IAM-Funktionen die E-ID betreffend geben. Klar, für den Speicherplatz selber braucht es ein Login, aber das muss nicht exorbitant integer sein (schwache 2FA sollte reichen), weil die verschlüsselte Ablage den Rest erledigt. Aber das alles regelt nur die Fragen von Backup&Recovery - der Einsatz muss immer noch lokal beim Anwender/Inhaber erfolgen, sonst hab ich ja wieder einen IdP (halt nur mehrere davon)....

**Comment by ii-nik** [comment]
Nein, warum? Die Cloud Wallet hat nichts mit dem Issuer zu tun. Ich kann mein Wallet-Anbieter wählen, dem ich vetraue. Zudem könnte man es so einrichten, dass die Daten erst im Agenten entschlüsselt werden.

**Reply by KampfCaspar** [reply]
Auch ohne die Cleartext-Daten - nur mit den 'Rahmendaten' - ist das gefährlich.

**Reply by grt-dti** [reply]
wenn die Wallet in der Cloud liegt, dann sind wir ja fast schon wieder beim IdP-Ansatz - halt ein anderes Föderationsverfahren, aber die meisten Gründe gegen den IdP würden hier auch greifen.
Erst wenn der Cloud-Anbieter ausschliesslich ein Storage-Provider ist (also nur zur Ablage von -selbstverständlich höchst-stark verschlüsseltem- Content), verpuffen die Gründe wieder. ABER dann laufen wir wieder in das Problem der Verlässlichkeit/Integrität/Diebstahlschutz......

**Reply by fforootd** [reply]
> wenn die Wallet in der Cloud liegt, dann sind wir ja fast schon wieder beim IdP-Ansatz - halt ein anderes Föderationsverfahren, aber die meisten Gründe gegen den IdP würden hier auch greifen.

Absolut, mir stellt sich dabei auch die Frage wer die Verantwortung für Impersonation übernimmt, sei es durch technische Fehler oder Angriffe.

**Reply by grt-dti** [reply]
haha; das ist ein generelles Problem in diesem System..... Wer übernimmt die Verantwortung, wenn in diesem ganzen heillos komplexen System etwas schief geht? Die Integrität ist erst dann da, wenn alle Komponenten (von der Identifikation des Inhabers und der Ausstellung des VC über die sichere Speicherung auf dem Smartphone bis zum datensparsamen Einsatz entlang festgelegter Regeln) harmonisch integer sind. Wenn das Wallet oder das Smartphone "Mist" baut, ist alles beim Teufel und wer trägt dann die Verantwortung und das Schadensrisiko?
Ich (privat) will dieses Risiko nicht tragen und ich wage zu postulieren, 50% der Schweizer könnten es ähnlich sehen. Wenn der Staat von mir erwartet, eine E-ID mit mir rumzutragen (z.B. weil mein Covid-Zertifikat daran gebunden ist), dann erwarte ich, dass mir dieser Staat auch nur die für mich verständlichen und von mir mitigierbaren Restrisiken überbordet und ansonsten eine unter diesen Restrisiken sichere Lösung bereitstellt - so ist es beim "Plastikausweis" und so muss es auch beim "elektronischen Ausweis" sein. Genau aus diesem Spannungsfeld (neben einigen wenigen Anderen) wird sich das allfällige Referendum speisen und ich hab aktuell noch keine Idee, wie man dem begegnen kann. Mein Gefühl sagt nur, SSI kann das heute noch nicht leisten, weil es in den relevanten Köpfen des Souveräns noch nicht im Ansatz angekommen ist.....

**Reply by fforootd** [reply]
Eines der grössten Probleme mit VC sehe ich btw. darin, dass beim verifier das Risiko besteht die beglaubigten Daten zu "leaken" und so ein Haufen "wertvoller" Personendaten in den Umlauf kommt, welche mit der proof einen signifikant höheren Wert haben als ohne.

ZKP kann das nur zum Teil lösen 😁
